lib/active_interactor/context/status.rb

```diff
@@ -81,10 +81,22 @@ def resolve
       def rollback!
         return false if @_rolled_back
 
-        _called.reverse_each(&:rollback)
+        _called.reverse_each do |interactor|
+          next if interactor.options.skip_rollback
+
+          execute_interactor_rollback!(interactor)
+        end
         @_rolled_back = true
       end
 
+      def execute_interactor_rollback!(interactor)
```

Comment:
I think we definitely want to make this private as this will change in the 1.2.0 release (moving to state & result objects) and we don't want to introduce any new public methods in this area.

Reply:
I'm also not sure we want to add this to the context object at all, I don't know if I like having the context understanding even more than it already does about the interactor.

Reply:
perhaps we could instead make `_called` public (I know I just said no new public method 😑 ) and call it appropriately in the worker.  Additionally I wonder if the issue is rollback is being called on the individual interactors instead of the organizer.  A possible fix for this is to pass `skip_rollback` to all the interactors in an organizer and just ensure `skip_rollback` IS called on the organizer which should have all called interactors in it's context.

Reply:
https://github.com/aaronmallen/activeinteractor/blob/main/lib/active_interactor/organizer/perform.rb#L78

```ruby
result = execute_interactor_with_callbacks(interface, true, skip_rollback: true)
```

Reply:
I totally agree it should be private and It also feels like the wrong place for this logic. I was just validating that it is the root cause.

Reply:
To handle this with a better approach requires some refactoring. Maybe we can move that to 2.0? WDYT ?

Reply:
what methods do you think we need to break to get this working properly?

Reply:
We need to
* Pass `skip_rollback` to child interactors
* Make `_called` public
* `execute_rollback_with_callbacks` and  and `execute_rollback` from worker

Reply:
I tend to agree, I think we can address this is `v1.2.0` when we make the switch over to state and result objects, as we'll be marking `ActiveInteractor::Context::Status` as deprecated in that version.